src/sentry/integrations/github/integration.py

```diff
@@ -267,29 +268,49 @@ def dispatch(self, request: Request, pipeline: Pipeline) -> Response:
             pipeline.bind_state("reinstall_id", request.GET["reinstall_id"])
 
         if "installation_id" in request.GET:
+            organization = self.get_active_organization(request)
+
+            # We want to wait until the scheduled deletions finish or else the post install to migrate repos do not work.
+            integration_pending_deletion_exists = OrganizationIntegration.objects.filter(
+                integration__provider=GitHubIntegrationProvider.key,
+                organization=organization,
+                status__in=[ObjectStatus.PENDING_DELETION],
+            ).exists()
+
+            if integration_pending_deletion_exists:
+                context = {
+                    "payload": {
+                        "success": False,
+                        "data": {"error": _("GitHub installation pending deletion.")},
+                    }
+                }
+                return render_to_response(
+                    "sentry/integrations/integration-pending-deletion.html",
+                    context=context,
+                    request=request,
+                )
```

Comment:
nit: If `context` is only used this once and has the same name as the kwarg, you can probably just move it all into the return statement
```python
return render_to_response(
    "sentry/integrations/integration-pending-deletion.html",
    context={
        "payload": {
             "success": False,
             "data": {"error": _("GitHub installation pending deletion.")},
         }
    },
    request=request,
)
```